coordinator/src/node.rs

```diff
@@ -403,3 +415,90 @@ pub fn process_incoming_messages_internal(
 
     Ok(())
 }
+
+#[cfg(test)]
+pub mod tests {
```

Comment:
👍 You might want to look into `proptest` or `quickcheck` to generate enough test cases to give you more confidence in the properties you are checking.